src/coreclr/System.Private.CoreLib/src/System/String.CoreCLR.cs

```diff
@@ -9,6 +10,20 @@ namespace System
 {
     public partial class String
     {
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "String_StrCns")]
+        private static unsafe partial void* StrCnsInternal(uint rid, IntPtr scopeHandle);
```

Comment:
I assume `StringHandleOnStack` is too expensive here?

Reply:
The issue is I need to dereference this pointer in COOP mode so either I (A) switch to COOP mode and do the pointer deref in C++ or (B) return to COOP naturally and use the `Unsafe.Ref()` API. I prefer the latter as it does managed object manipulation in C#.

Reply:
Is there any concern about the liveness of `LoaderAllocator`?

Can a GC happen between before the dereference and cause the LoaderAllocator become dead?

Reply:
> Is there any concern about the liveness of LoaderAllocator?

Nope. The `scopeHandle` passed in represents the module being operated on. It must be assumed that the caller is extending the lifetime of the module while passing an opaque handle to it.